### ISSUE TITLE
Trying to understand how the colorization works...

### ISSUE THREAD

**Issue by mpenna** [ekonbenefits/vscode-harbour#2]
Congratulations for this nice extension. I've been fiddling with the .tmLanguage file a bit and by now I kind of understand how to edit/insert new keywords and such, but I haven't been able to grasp (and ultimately customize) the colorization aspects of the language. Could you offer some insights on how language and colorization (theme?) go along and work together?

**Comment by jbtule** [comment]
[VS Code Colorizer Docuemntation](https://code.visualstudio.com/docs/customization/colorizer)
which referes to [Textmate Grammar Documentation](https://manual.macromates.com/en/language_grammars)

**Comment by mpenna** [comment]
@jbtule Thanks. I had already checked those docs before I decided to post here. Either I'm plain dumb or they're just not comprehensive enough. Nowhere on the harbour.tmLanguage file, or any other file in the extension's bundle, is there any information relating to color, right? So this surely is being defined someplace else. But where, then? Only thing that appears to have any relation to colorization is the "name" (scope selectors?) given to the patterns being defined inside the harbour.tmLanguage file. But being this such an "edge-case language" (I installed this extension mainly because it's the closest one I could find that would give me some colorization for an Xbase++ project I'm working on), something tells me that VS Code itself has no idea how to work with "harbour scopes". Given that VS Code doesn't come with this language pre-installed -- after all, that's the reason we all installed this package --, when it comes to affording a certain color to, say, a "numeric constant" on a *.PRG or *.CH file inside the editor, something must be determining that... I just can't figure out what or where. The current selected theme, perhaps? But, again, would "constant.numeric.harbour" mean anything to a theme that probably has no clue to what "harbour" means? It's all just a bit confusing, I should say.
